Detectors/GlobalTracking/src/MatchTOF.cxx

```diff
@@ -898,7 +898,10 @@ void MatchTOF::selectBestMatches()
       }
       if (!labelOk) {
         // we have not found the track label among those associated to the TOF cluster --> fake match! We will associate the label of the main channel, but negative
-        mOutTOFLabels.emplace_back(labelsTOF[0].getTrackID(), labelsTOF[0].getEventID(), labelsTOF[0].getSourceID(), true);
+	if(! labelsTOF.size())
+	  LOG(ERROR) << "TOF label not found since size of label is zero. This should not happen!!!!";
```

Comment:
Is'not it better to throw an exception here? Logging an error may go unnoticed.

Reply:
Do you mean a FATAL?

Reply:
sorry.
I guess you mean
 assert(labelsTOF.size());
I will add it.